csrc/ir/nodes.cpp

```diff
@@ -2625,10 +2627,14 @@ std::pair<IterDomain*, IterDomain*> IterDomain::split(
       factor->isIntegralScalar(), "Cannot split by non-integer value ", factor);
 
   // outer loop size
-  Val* remainder = ceilDiv(in->extent(), factor);
+  Val* remainder = getenv("SPLIT")
+      ? SimplifyingIrBuilder::ceilDivExpr(in->extent(), factor)
```

Comment:
I think I figured it out why this resulted in so many segfaults. Assuming there's no functional regression, is there any concern with this change? From my quick experiments, slice and concat with static shapes seem to be greatly simplified from the beginning, which is quite convenient for debugging.

However, it could be also considered less useful as you can't figure out how an iter domain is constructed by just looking at its extent. For example, `iS{ceilDiv(32, 4)}` would become just `iS{8}`. With the current representation, it's clear that it's an output of a size-32 domain split by a factor of 4, but that's not immediately visible just from the extent notation since it'd be just `8`. 

Overall, I'd prefer the simplification since it would also help avoid accumulating integer expressions around reshape. Wdyt? @zasdfgbnm @jacobhinkle

Reply:
> I think I figured it out why this resulted in so many segfaults. Assuming there's no functional regression, is there any concern with this change? 

I don't think so. We shouldn't be pattern matching on the extent formula anywhere to my knowledge.
 
> Overall, I'd prefer the simplification since it would also help avoid accumulating integer expressions around static resize. Wdyt? @zasdfgbnm @jacobhinkle

+1 I'm for simplification. I agree somewhat about the utility in scanning through the fusion math to check for split/merge but that also sometimes makes it tough to even see how many dimensions there are in a scheduled tensor.

Reply:
In my experience, I often feel not simplifying could be a problem. For example, in my mind, I can clearly remember that I am scheduling as `[..., 2, ....]`, but I can not remember and I don't really care how I get this `2` (for example, `split(split(I0, 128).outer, 2).inner` and `split(split(I0, 256).inner, 128).outer` is mathematically equivalent, so I don't care which one we are using). When I am looking for the `2` in the loop domain, it is most convenient that it is just `2`, but sometimes it can be complicated as `ceilDiv(ceilDiv(128, 8), 8)`, which takes me a few seconds to figure out if it is `2` or not.